### PR TITLE
WIP Open 3.2 and rerender

### PR DESCRIPTION
**Pull request by mariusvniekerk**
Probably not correct for windows / osx.   Seems to build on linux.

### CODE REVIEW

**Comment by jakirkham** [comment]
So there is PR ( https://github.com/conda-forge/opencv-feedstock/pull/50 ) for this already. That said, feel free to rebase on top of it and continue if you would like. Unfortunately it seems building a new release of OpenCV seems to be a Herculean effort each time; so, we end up passing the baton a few times. If you have time to push forward for a bit, the effort would definitely be appreciated.

**Comment by jakirkham** [comment]
Toggling to refresh CI given the update to `master`.

**Comment by jakirkham** [comment]
After toggling it appears that OpenCV is making use of C++11 features. So I don't know if it will be possible to support any Python versions older than 3.5 on Windows as VC 14 will be required. We could raise an issue with them to see if they will back off this requirement.

File: recipe/opencv_persistence_data_member.patch [hunk 1]
```diff
@@ -0,0 +1,199 @@
+--- modules/core/src/persistence.cpp
++++ modules/core/src/persistence.cpp
+@@ -1367,7 +1367,7 @@
+             CV_PARSE_ERROR("Unrecognized Base64 header");
+ 
+         std::vector<char> header(base64::HEADER_SIZE + 1, ' ');
+-        base64::base64_decode(beg, header.data(), 0U, base64::ENCODED_HEADER_SIZE);
++        base64::base64_decode(beg, &header.front(), 0U, base64::ENCODED_HEADER_SIZE);
```
Comment (jakirkham) [inline recipe/opencv_persistence_data_member.patch]: Actually it looks like our own patches are to blame. Could we please try dropping this patch, @mariusvniekerk? If we do need it, we will need to come up with something else.

Comment (jakirkham) [inline recipe/opencv_persistence_data_member.patch]: One thing we could try is changing `&header.front()` to `&(header[0])`. Similar changes could be repeated throughout this patch.

**Comment by jakirkham** [comment]
Alright I've tried pushing two commits to your branch @mariusvniekerk. They drop these two patches. Given my discussion with one of the OpenCV devs, it doesn't sound like they should be needed. At least this way we can figure out what happens if they are not included. If we do have new issues, we can try to raise those.

**Comment by mariusvniekerk** [comment]
@jakirkham we may need the matrix path for win and py27

**Comment by jakirkham** [comment]
Yep, I think you are right that we need both of these. After looking at what is going on a bit more, I think there was some confusion in the `persistence.cpp` patch. It appears some cases we have a `std::string` and others we have a `std::vector`. They differ in that `std::string` has `data`, but does not have `front` until C++11; whereas, `std::vector` is the exact opposite having `front`, but does not have `data` until C++11. The fact that `opencv` is using both to work with string-like data only confuses the matter. Have tried to update the patch to hopefully more correctly address these two cases.

**Comment by mariusvniekerk** [comment]
(or crazy thought we stop building opencv for py27 and windows).  There are other methods that users can use to get it installed on that.  Not sure if we want to keep the mammoth burden to maintain it.

**Comment by jakirkham** [comment]
The thing is none of these problems are actually particular to Windows Python 2.7. They are all cases where the OpenCV dev team has assumed C++11 support. AFAICT though they are not requiring this support in their CMake build scripts nor am I seeing it in the docs.

**Comment by patricksnape** [comment]
Yeah sorry, this is my fault - I had lazily replaced the `front` members without really understanding the code and so some are `vector` and some are `string` as @jakirkham mentioned. So we just need to fix that patch to properly replace some with front and others with the ugly dereference.

**Comment by jakirkham** [comment]
No worries, but we do have to patch more files. If you have time to contribute more patches, that would be great. FWICT it is the same story with the current round of build errors.

**Comment by patricksnape** [comment]
#50 now works, so we can close this

**Pull request closed without merging**